[rocThrust] Fix google benchmark compile issue on Windows (#643)
ROCm/rocm-libraries: projects/rocthrust/CHANGELOG.md

* Updated the required version of Google Benchmark from 1.8.0 to 1.9.0.
* Drop `c++14` support for rocthrust.
* Renamed `cpp14_required.h` to `cpp_version_check.h`

> * `clang++` is now used to compile google benchmarks on Windows.```

Reviewer: There's an extra three backticks at the end of line 13. I won't hold back the change because of this, but it might be nice to fix if we can.

Author: I will merge this in since if I change it now it will get rid of all the approvals. I will fix this in the code coverage PR (#410)

> * Switched to using `clang++` to compile for google benchmark on Windows system

Reviewer 2: ```suggestion
* `clang++` is now used to compile google benchmarks on Windows.```
